src/app/application.cpp

```diff
@@ -771,28 +767,43 @@ void Application::shutdownCleanup(QSessionManager &manager)
 }
 #endif
 
-#ifdef Q_OS_WIN
 void Application::applyMemoryWorkingSetLimit()
 {
-    const SIZE_T UNIT_SIZE = 1024 * 1024; // MiB
-    const SIZE_T maxSize = memoryWorkingSetLimit() * UNIT_SIZE;
-    const SIZE_T minSize = std::min<SIZE_T>((64 * UNIT_SIZE), (maxSize / 2));
+    const size_t MiB = 1024 * 1024;
+    const QString logMessage = tr("Failed to set physical memory (RAM) usage limit. Error code: %1. Error message: \"%2\"");
+
+#ifdef Q_OS_WIN
+    const SIZE_T maxSize = memoryWorkingSetLimit() * MiB;
+    const SIZE_T minSize = std::min((64 * MiB), (maxSize / 2));
     if (!::SetProcessWorkingSetSizeEx(::GetCurrentProcess(), minSize, maxSize, QUOTA_LIMITS_HARDWS_MAX_ENABLE))
     {
         const DWORD errorCode = ::GetLastError();
         QString message;
         LPVOID lpMsgBuf = nullptr;
-        if (::FormatMessageW((FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS)
-                         , nullptr, errorCode, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), reinterpret_cast<LPWSTR>(&lpMsgBuf), 0, nullptr))
+        const DWORD msgLength = ::FormatMessageW((FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS)
+            , nullptr, errorCode, LANG_USER_DEFAULT, reinterpret_cast<LPWSTR>(&lpMsgBuf), 0, nullptr);
+        if (msgLength > 0)
         {
             message = QString::fromWCharArray(reinterpret_cast<LPWSTR>(lpMsgBuf)).trimmed();
             ::LocalFree(lpMsgBuf);
         }
-        LogMsg(tr("Failed to set physical memory (RAM) usage limit. Error code: %1. Error message: \"%2\"")
-               .arg(QString::number(errorCode), message), Log::WARNING);
+        LogMsg(logMessage.arg(QString::number(errorCode), message), Log::WARNING);
+    }
+#elif defined(Q_OS_UNIX)
+    // has no effect on linux but it might be meaningful for other OS
```

Comment:
I think something similar should be shown in UI.

Reply:
Added tooltip for the option.